datafusion/core/src/test/mod.rs

```diff
@@ -102,11 +105,10 @@ pub fn partitioned_file_groups(
     partitions: usize,
     file_type: FileType,
     file_compression_type: FileCompressionType,
+    work_dir: &Path,
```

Comment:
are Temporary directories created by `TempDir::new` are deleted in `TempDir::drop`.
But files created in the temporary directory can be refered by callers of this method, so we can't call `TempDir::new` in this method.
The life cycle of such temporary directories should be managed by callers.